### PR TITLE
Add new topic for conflict detection

### PR DESCRIPTION
**Pull request by hbc-bindu**
### What does this PR do?

-- Create a new topic for conflict detection feature 
-- Update the (existing) development models 
-- Update the side bar

### What issues does this PR fix or reference?
#<Insert GitHub Issue>, @<Insert GUS WI>@

### Functionality Before
<insert gif and/or summary>

### Functionality After
<insert gif and/or summary>

### CODE REVIEW

File: docs/_articles/en/user-guide/detect-conflicts.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: Detect Conflicts
+lang: en
+---
+
+Whenever you deploy or retrieve source to keep the local project and your default org in sync, the operation could detect conflicts. If conflicts are not detected, the retrieve or deploy operation is completed. If conflicts are detected, the retrieve or deploy operation aborts. The resource files that caused the conflict, along with the CLI command to overwrite conflicts are displayed in the Output panel.
```
Comment (lcampos) [inline docs/_articles/en/user-guide/detect-conflicts.md]: I think we should make it clear this feature is only for non-scratch orgs. cc @hbc-bindu @smaddox-sf @rcoringrato-sfdc

Comment (hbc-bindu) [reply docs/_articles/en/user-guide/detect-conflicts.md]: Done. Added:
You can enable the conflict detection feature only in non-scratch orgs.

Comment (rcoringrato-sfdc) [reply docs/_articles/en/user-guide/detect-conflicts.md]: Should we change: "If conflicts are detected, the retrieve or deploy operation aborts." to "If conflicts are detected, the retrieve or deploy operation can be cancelled."?
@hbc-bindu @lcampos

Comment (lcampos) [reply docs/_articles/en/user-guide/detect-conflicts.md]: @rcoringrato-sfdc right now we completely cancel the operation until the user manually fixes the conflicts and then they need to re-run the command (deploy/retireve) right ? If so I think cancelled might be more in line with how we talk about deploy/retrieve operations.

Comment (rcoringrato-sfdc) [reply docs/_articles/en/user-guide/detect-conflicts.md]: @lcampos The current conflict warning dialog presents the options to either Cancel, or to Override Conflicts. Choosing to override allows the initial operation to continue.

Comment (rcoringrato-sfdc) [reply docs/_articles/en/user-guide/detect-conflicts.md]: @lcampos @smaddox-sf 
I've updated the doc, and images.

**Pull request merged**